closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -472,46 +502,60 @@ int decryptFinal(byte[] in, int inOfs, int len,
         if (len < 0) {
             throw new ProviderException("Input length is negative");
         }
-        if (len < tagLenBytes - ibuffer.size()) {
-            throw new AEADBadTagException("Input too short - need tag");
-        }
-        if (len > MAX_BUF_SIZE - ibuffer.size()) {
-            throw new ProviderException("SunJCE provider only supports "
-                + "a positive input size up to " + MAX_BUF_SIZE + " bytes");
-        }
 
-        if (out.length - outOfs < len  + ibuffer.size() - tagLenBytes) {
-            throw new ShortBufferException("Output buffer too small");
-        }
+        byte[] keyCopy;
+        byte[] ivCopy;
+        byte[] aad;
+        int localTagLenBytes;
+        int ret = 0;
```

Comment:
Please remove the redundant initialization here.

Reply:
Removed.